File: cc_bot.py
import requests
import json

bot_token = "8354463122:AAF9nR5ePOHFdGYPUPglqypAraar-CqH6PY"
checker_url = "https://test.infinitemsfeed.com/chk.php?lista="

BOT_CREDIT = "@abinjmoffical"


def send_message(chat_id, text, reply_to=None):
    url = f"https://api.telegram.org/bot{bot_token}/sendMessage"
    params = {
        "chat_id": chat_id,
        "text": text,
        "parse_mode": "HTML"
    }
    if reply_to:
        params["reply_to_message_id"] = reply_to

    r = requests.get(url, params=params)
    return r.json().get("result", {}).get("message_id")


def edit_message(chat_id, message_id, new_text):
    url = f"https://api.telegram.org/bot{bot_token}/editMessageText"
    params = {
        "chat_id": chat_id,
        "message_id": message_id,
        "text": new_text,
        "parse_mode": "HTML"
    }
    requests.get(url, params=params)


def process_update(update):
    if "message" not in update:
        return

    chat_id = update["message"]["chat"]["id"]
    message_id = update["message"]["message_id"]
    text = update["message"].get("text", "")

    if not text.startswith("/chk"):
        return

    wait_msg = send_message(chat_id, "Checking... Card", message_id)

    cc = text.replace("/chk", "").strip()
    if not cc:
        edit_message(chat_id, wait_msg, "Please provide a valid card.")
        return

    try:
        response_api = requests.get(checker_url + cc, verify=False, timeout=15)
        result = response_api.json()
    except Exception as e:
        result = {"error": str(e)}

    response = ""

    if "success" in result:

        if result["success"] is True:
            response += "𝗦𝘁𝗮𝘁𝘂𝘀 - Approved ✅\n"
            response += "━━━━━━━━━━━━━\n"
            response += f"[ϟ] 𝗖𝗖 ⌁ {cc}\n"
            response += f"[ϟ] 𝗦𝘁𝗮𝘁𝘂𝘀 : {result.get('message', 'Approved')} ✅\n"
            response += "[ϟ] 𝗚𝗮𝘁𝗲 - Stripe Auth\n"
            response += "━━━━━━━━━━━━━\n"
            response += f"[ϟ] 𝗦𝗰𝗿𝗮𝗽𝗽𝗲𝗱 𝗕𝘆 : {BOT_CREDIT}\n"

        else:
            response += "𝗦𝘁𝗮𝘁𝘂𝘀 - 𝘿𝙀𝘾𝙇𝙄𝙉𝙀𝘿 ❌\n"
            response += "━━━━━━━━━━━━━\n"
            response += f"[ϟ] 𝗖𝗖 ⌁ {cc}\n"
            response += f"[ϟ] 𝗦𝘁𝗮𝘁𝘂𝘀 : {result.get('message', 'Declined')} ❌\n"
            response += "[ϟ] 𝗚𝗮𝘁𝗲 - Stripe Auth\n"
            response += "━━━━━━━━━━━━━\n"
            response += f"[ϟ] 𝗦𝗰𝗿𝗮𝗽𝗽𝗲𝗱 𝗕𝘆 : {BOT_CREDIT}\n"

    else:
        response += "𝗦𝘁𝗮𝘁𝘂𝘀 - ERROR ❌\n"
        response += "━━━━━━━━━━━━━\n"
        response += f"[ϟ] X Error: {result.get('error', 'Unknown error')}\n"

    edit_message(chat_id, wait_msg, response)


# Example test
update = json.loads(
    '{"message": {"chat": {"id": 123456789}, "message_id": 1, "text": "/chk 1234567890123456"}}'
)
process_update(update)

File: cc_bot.php
<?php

$botToken = "8354463122:AAF9nR5ePOHFdGYPUPglqypAraar-CqH6PY";
$checkerUrl = "https://test.infinitemsfeed.com/chk.php?lista=";

$update = json_decode(file_get_contents('php://input'), true);

if(isset($update['message'])){
    $chatId = $update['message']['chat']['id'];
    $message = $update['message']['message_id'];
    $text = $update['message']['text'];

    if(strpos($text, '/chk') === 0){
        $waitMsg  = sendMessage($chatId, "Checking... Card", $messageId);

        $cc = trim(str_replace('/chk ', '', $text));

        if(empty($cc)){
            editMessage($chatId, $waitMsg, "Please provide a valid card number");
            exit();
        }

        $ch = curl_init();
        curl_setopt($ch, CURLOPT_URL, $checkerUrl . $cc);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

        $result = curl_exec($ch);
        curl_close($ch);

        $result = json_decode($result, true);

        $response = "Card: " . $cc . "\n\n";
if (isset($result['success'])) {

    if ($result['success'] === true) {

        $response .= "𝗦𝘁𝗮𝘁𝘂𝘀 - Approved ✅\n";
        $response .= "━━━━━━━━━━━━━\n";
        $response .= "[ϟ] 𝗜𝗡𝗣𝗨𝗧 ⌁ {$value}\n";
        $response .= "[ϟ] 𝗦𝘁𝗮𝘁𝘂𝘀 : " . ($result['message'] ?? "Approved") . " ✅\n";
        $response .= "[ϟ] 𝗚𝗮𝘁𝗲 - API Check\n";
        $response .= "━━━━━━━━━━━━━\n";
        $response .= "[ϟ] 𝗦𝗰𝗿𝗮𝗽𝗽𝗲𝗱 𝗕𝘆 : @abinjmoffical\n";

    } else {

        $response .= "𝗦𝘁𝗮𝘁𝘂𝘀 - 𝘿𝙀𝘾𝙇𝙄𝙉𝙀𝘿 ❌\n";
        $response .= "━━━━━━━━━━━━━\n";
        $response .= "[ϟ] 𝗜𝗡𝗣𝗨𝗧 ⌁ {$value}\n";
        $response .= "[ϟ] 𝗦𝘁𝗮𝘁𝘂𝘀 : " . ($result['message'] ?? "Declined") . " ❌\n";
        $response .= "[ϟ] 𝗚𝗮𝘁𝗲 - API Check\n";
        $response .= "━━━━━━━━━━━━━\n";
        $response .= "[ϟ] 𝗦𝗰𝗿𝗮𝗽𝗽𝗲𝗱 𝗕𝘆 : @abinjmoffical\n";
    }

} else {

    $response .= "𝗦𝘁𝗮𝘁𝘂𝘀 - ERROR ❌\n";
    $response .= "━━━━━━━━━━━━━\n";
    $response .= "[ϟ] X Error: " . ($result['error'] ?? "Unknown error") . "\n";
}

        editMessage($chatId, $waitMsg, $response);

    }
}

function sendMessage($chatId, $text, $replyTo = null){
    global $botToken;
    $params = [
        'chat_id' => $chatId,
        'text' => $text,
        'parse_mode' => 'HTML',
    ];

if($replyTo){
    $params['reply_to_message_id'] = $replyTo;
}

$response = file_get_contents("https://api.telegram.org/bot$botToken/sendMessage?" . http_build_query($params));
$response = json_decode($response, true);

return $response['result']['message_id'] ?? null;



}


function editMessage($chatId, $messageId, $newText){
    global $botToken;
    $params = [
        'chat_id' => $chatId,
        'message_id' => $messageId,
        'text' => $newText,
        'parse_mode' => 'HTML',
    ];

   file_get_contents("https://api.telegram.org/bot$botToken/editMessageText?" . http_build_query($params));
}


